galaxy_ng/app/api/ui/v2/permissions.py

```diff
@@ -16,3 +20,77 @@ def has_permission(self, request, view):
             return True
         # Otherwise, deny access
         return False
+
+
+class ComplexUserPermissions(AnsibleBaseUserPermissions):
+    """
+    ComplexUserPermissions complies with the "complex" requirements
+    of a system where there is a resource server that syncs users
+    and a jwt that creates users and a confusing mix of how is_superuser
+    can get populated on galaxy when ALLOW_LOCAL_RESOURCE_MANAGEMENT
+    is False,
+
+    To try to break down the logic when ALLOW_LOCAL_RESOURCE_MANAGEMENT=False:
+        - users CAN NOT be created/edited/deleted directly in galaxy, except ...
+        - if the request is a PATCH or a PUT that is only modifying the value
+          of is_superuser, then it can be allowed
+        - if the caller is a superuser, they can set the is_superuser flag for
+          a user to True or False
+        - if the caller is the user being modified, they can only set the flag
+          to False
+        - No user can be demoted from superuser when they are the last superuser
```

Comment:
Note that I don't think we need this condition, because it could be that no superusers are found in the local system, but JWT could make a user a superuser same-request later on.